### PR TITLE
Add popup timeout config

### PR DESCRIPTION
**Pull request by luisrudge**
fix https://github.com/auth0/auth0-spa-js/issues/92

### Description

Adds a config object to send popup configuration options. For now, only `timeoutInSeconds` is available.

### References

https://github.com/auth0/auth0-spa-js/issues/92

### Testing

- [x] This change adds test coverage for new/changed/fixed functionality

### CODE REVIEW

**Comment by mwillbanks** [comment]
Looks great to me.

File: __tests__/utils.test.ts [hunk 1]
```diff
@@ -297,13 +297,33 @@ describe('utils', () => {
...
       expect(popup.close).toHaveBeenCalled();
     });
-    it('times out after 60s', async () => {
+    it('times out after config.timeoutInSeconds', async () => {
+      const { popup, url } = setup('');
+      const seconds = 10;
+      /**
+       * We need to run the timers after we start `runPopup`, but we also
+       * need to use `jest.useFakeTimers` to trigger the timeout.
+       * That's why we're using a real `setTimeout`, then using fake timers
```
Comment (lbalmaceda) [inline __tests__/utils.test.ts]: image

File: src/utils.ts [hunk 2]
```diff
@@ -54,12 +54,16 @@ export const openPopup = () => {
...
+  authorizeUrl: string,
+  config: PopupConfigOptions
+) => {
   popup.location.href = authorizeUrl;
   return new Promise<AuthenticationResult>((resolve, reject) => {
     const timeoutId = setTimeout(() => {
-      reject(TIMEOUT_ERROR);
-    }, 60 * 1000);
+      reject({ ...TIMEOUT_ERROR, popup });
+    }, config.timeoutInSeconds || 60 * 1000);
```
Comment (lbalmaceda) [inline src/utils.ts]: I see you are using `60` here and in `src/Auth0Client.ts`. Please, find a way to reuse this definition, either via a constant or file.

File: src/global.ts [hunk 3]
```diff
@@ -117,6 +117,9 @@ interface RedirectLoginResult {
 }
 
 interface PopupLoginOptions extends BaseLoginOptions {}
+interface PopupConfigOptions {
```
Comment (lbalmaceda) [inline src/global.ts]: docs please, anywhere? Maybe on the README?

File: src/utils.ts (hunk 2 above)
Comment (luisrudge) [reply src/utils.ts]: done

File: src/global.ts (hunk 3 above)
Comment (luisrudge) [reply src/global.ts]: done

File: __tests__/index.test.ts [hunk 4]
```diff
@@ -32,6 +32,10 @@ const TEST_TELEMETRY_QUERY_STRING = `&auth0Client=${encodeURIComponent(
   )
 )}`;
 
+const DEFAULT_POPUP_CONFIG_OPTIONS: PopupConfigOptions = {
+  timeoutInSeconds: 60
```
Comment (lbalmaceda) [inline __tests__/index.test.ts]: Shouldn't this be pointing to the constant file as well?
```
import { DEFAULT_AUTHORIZE_TIMEOUT_IN_SECONDS } from '../src/constants';

```

File: src/global.ts [hunk 5]
```diff
@@ -118,6 +118,14 @@ interface RedirectLoginResult {
 
 interface PopupLoginOptions extends BaseLoginOptions {}
 
+interface PopupConfigOptions {
+  /**
+   * The number of seconds to wait for a popup response before
+   * throwing a timeout error
```
Comment (lbalmaceda) [inline src/global.ts]: Please add the default value, or point to the constant name for the user to check if you think this will change often.

**Pull request merged**